internal/grpcserver/storage/elastic.go

```diff
@@ -0,0 +1,213 @@
+package storage
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"github.com/astanishevskyi/grpc-server/internal/grpcserver/models"
+	elastic "github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type ElasticStorage struct {
+	client *elastic.Client
+	mu     sync.Mutex
+	lastID uint32
+}
+
+func NewElasticStorage() *ElasticStorage {
+	user := os.Getenv("ELASTIC_USER")
```

Comment:
as a common practice there should be an `env` or `environment` package. It is the place where you deal with all env variables and flags. Btw, you could check out `viper` package) So, this method should be refactored like this:
```golang
func NewElasticStorage(user, pass, addr string) *ElasticStorage {
```

Reply:
The main idea of this would be to decouple the obtaining of config variables and using them. So that you could use the `ElasticStorage` anywhere and provide configuration, specific to your application way.